shark-graph/src/test/java/shark/HprofDeobfuscatorTest.kt

```diff
@@ -0,0 +1,148 @@
+          .map { it.name }
+          .toList()
+      ).contains(MONEY_STATIC_FIELD_NAME)
+    }
+  }
+
+  private fun File.writeRecords(
+    records: List<HprofRecord>
+  ) {
+    HprofWriter.open(this)
+      .use { writer ->
+        records.forEach { record ->
+          writer.write(record)
+        }
+      }
+  }
+
+  private fun File.readHprof(block: (HeapGraph) -> Unit) {
+    Hprof.open(this)
+      .use { hprof ->
+        block(HprofHeapGraph.indexHprof(hprof))
+      }
+  }
+
+  private fun createRecords(): List<HprofRecord> {
+    val obfuscatedWalletClassName = StringRecord(id, OBFUSCATED_WALLET_CLASS_NAME)
+    val obfuscatedMoneyClassName = StringRecord(id, OBFUSCATED_MONEY_CLASS_NAME)
+    val obfuscatedPennyInstanceFieldName = StringRecord(id, OBFUSCATED_PENNY_INSTANCE_FIELD_NAME)
+    val obfuscatedMoneyStaticFieldName = StringRecord(id, OBFUSCATED_MONEY_STATIC_FIELD_NAME)
+
+    val loadObfuscatedMoneyClass = LoadClassRecord(1, id, 1, obfuscatedMoneyClassName.id)
+    val loadObfuscatedWalletClass = LoadClassRecord(1, id, 1, obfuscatedWalletClassName.id)
+
+    val obfuscatedMoneyClassDump = ClassDumpRecord(
+      id = loadObfuscatedMoneyClass.id,
+      stackTraceSerialNumber = 1,
+      superclassId = 0,
+      classLoaderId = 0,
+      signersId = 0,
+      protectionDomainId = 0,
+      instanceSize = 0,
+      staticFields = listOf(),
+      fields = listOf(
+        FieldRecord(obfuscatedPennyInstanceFieldName.id, PrimitiveType.INT.hprofType)
+      )
+    )
+
+    val obfuscatedMoneyInstanceDump = InstanceDumpRecord(
+      id = id,
+      stackTraceSerialNumber = 1,
+      classId = loadObfuscatedMoneyClass.id,
+      fieldValues = byteArrayOf(0x0, 0x1, 0x2, 0x3)
+    )
+
+    val obfuscatedWalletClassDump = ClassDumpRecord(
+      id = loadObfuscatedWalletClass.id,
+      stackTraceSerialNumber = 1,
+      superclassId = 0,
+      classLoaderId = 0,
+      signersId = 0,
+      protectionDomainId = 0,
+      instanceSize = 0,
+      staticFields = listOf(
+        StaticFieldRecord(
+          obfuscatedMoneyStaticFieldName.id, PrimitiveType.REFERENCE_HPROF_TYPE,
+          ReferenceHolder(obfuscatedMoneyInstanceDump.id)
+        )
+      ),
+      fields = emptyList()
+    )
+
+    return listOf(
+      obfuscatedWalletClassName, obfuscatedMoneyClassName, obfuscatedPennyInstanceFieldName,
+      obfuscatedMoneyStaticFieldName, loadObfuscatedMoneyClass, loadObfuscatedWalletClass,
+      obfuscatedMoneyClassDump, obfuscatedMoneyInstanceDump, obfuscatedWalletClassDump
+    )
+  }
+
+  companion object {
+    const val WALLET_CLASS_NAME = "Wallet"
+    const val MONEY_CLASS_NAME = "Money"
+    const val MONEY_STATIC_FIELD_NAME = "money"
+    const val PENNY_INSTANCE_FIELD_NAME = "penny"
+
+    const val OBFUSCATED_WALLET_CLASS_NAME = "a"
+    const val OBFUSCATED_MONEY_CLASS_NAME = "b"
+    const val OBFUSCATED_MONEY_STATIC_FIELD_NAME = "c.d"
+    const val OBFUSCATED_PENNY_INSTANCE_FIELD_NAME = "e.f"
```

Comment:
As the surface increases in tests, you might find yourself with a lot of constants. Another option here might be to write something that obfuscates in a dumb way.